launchpad/src/launchpad.rs

```diff
@@ -1,9 +1,489 @@
+pub trait Launchpad: setup::SetupModule + ongoing_operation::OngoingOperationModule {
+    // endpoints - owner-only
+
+    #[only_owner]
+    #[endpoint(claimTicketPayment)]
+    fn claim_ticket_payment(&self) -> SCResult<()> {
+        let ticket_payment_token = self.ticket_payment_token().get();
+        let sc_balance = self.blockchain().get_sc_balance(&ticket_payment_token, 0);
+        let owner = self.blockchain().get_caller();
+
+        if sc_balance > 0 {
+            self.send()
+                .direct(&owner, &ticket_payment_token, 0, &sc_balance, &[]);
+        }
+
+        Ok(())
+    }
+
+    #[only_owner]
+    #[endpoint(addTickets)]
+    fn add_tickets(
+        &self,
+        #[var_args] address_number_pairs: VarArgs<MultiArg2<Address, usize>>,
+    ) -> SCResult<()> {
+        self.require_no_ongoing_operation()?;
+
+        let current_epoch = self.blockchain().get_block_epoch();
+        let winner_selection_start_epoch = self.winner_selection_start_epoch().get();
+        require!(
+            current_epoch < winner_selection_start_epoch,
+            "Cannot add more tickets, winner selection has started"
+        );
+
+        for multi_arg in address_number_pairs.into_vec() {
+            let (buyer, nr_tickets) = multi_arg.into_tuple();
+
+            self.try_create_tickets(&buyer, nr_tickets)?;
+        }
+
+        Ok(())
+    }
+
+    // endpoints
+
+    #[endpoint(selectWinners)]
+    fn select_winners(&self) -> SCResult<OperationCompletionStatus> {
+        require!(
+            self.confirmation_period_start_epoch().is_empty(),
+            "Cannot select winners after confirmation period started"
+        );
+
+        let current_epoch = self.blockchain().get_block_epoch();
+        let winner_selection_start_epoch = self.winner_selection_start_epoch().get();
+
+        require!(
+            winner_selection_start_epoch >= current_epoch,
+            "Cannot select winners yet"
+        );
+
+        let last_ticket_position = self.shuffled_tickets().len();
+        let nr_winning_tickets = self.nr_winning_tickets().get();
+        let ongoing_operation = self.current_ongoing_operation().get();
+
+        // dummy, will be overwritten in the Load part, 
+        // but the Rust compiler complains of possibly uninitialized variables otherwise
+        let mut rng = Random::from_hash(self.crypto(), H256::zero(), 0);
+        let mut ticket_position = 0;
+
+        self.run_while_it_has_gas(ongoing_operation, |gas_op| match gas_op {
+            GasOp::Load(op) => match op {
+                OngoingOperationType::None => {
+                    rng = Random::from_seeds(
+                        self.crypto(),
+                        self.blockchain().get_prev_block_random_seed(),
+                        self.blockchain().get_block_random_seed(),
+                    );
+                    ticket_position = VEC_MAPPER_START_INDEX;
+
+                    LoopOp::Continue
+                }
```

Comment:
This code is unreachable because of the match inside `run_while_it_has_gas` which filters out `OngoingOperationType::None`.
If we keep the initialization here, then the match in the `run_while_it_has_gas` should be removed.

Reply:
Alternatively, set the proper defaults above.